x-pack/plugins/integration_assistant/public/services/experimental_features_service.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExperimentalFeatures } from '../../common/experimental_features';
+
+export class ExperimentalFeaturesService {
+  private static experimentalFeatures?: ExperimentalFeatures;
+
+  public static init(experimentalFeatures: ExperimentalFeatures) {
```

Comment:
If we do not want to instantiate this service , may be a private constructor? Not sure TS allows this though.

Reply:
We do instantiate it in the plugin.ts

Reply:
You mean it calls the static `init` function right? I thought if `new ExperimentalFeaturesService()` is never called , then we may have a private constructor to avoid instantiation otherwise..

Reply:
Ah I see, I think I understand. I attempted to add a private constructor and it told me to remove it as it was unused

(FWIW all the other plugins with experimental features with this have almost this exact same file)